notebooks/model_conversion-pca.ipynb

```diff
@@ -0,0 +1,456 @@
+      "Do sklearn and torch KernelPCA models generate close results?\n",
+      "True\n",
+      "Do sklearn and torch TruncatedSVD models generate close results?\n",
+      "True\n"
+     ]
+    }
+   ],
+   "source": [
+    "for i, model_name in enumerate([\"PCA\", \"KernelPCA\", \"TruncatedSVD\"]):\n",
+    "    sklearn_transformed = sklearn_models[0].transform(X_test)\n",
+    "    torch_transformed = torch_models[0].transform(X_test)\n",
+    "    print(\"Do sklearn and torch \"+model_name+\" models generate close results?\")\n",
+    "    print(np.allclose(sklearn_transformed, torch_transformed, atol=1e-04))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c7c679b8",
+   "metadata": {
+    "papermill": {
+     "duration": 0.006634,
+     "end_time": "2023-04-03T08:30:21.699490",
+     "exception": false,
+     "start_time": "2023-04-03T08:30:21.692856",
+     "status": "completed"
+    },
+    "tags": []
+   },
+   "source": [
+    "# Convert the models to onnx using skl2onnx and test the results"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "690dc468",
+   "metadata": {
+    "execution": {
+     "iopub.execute_input": "2023-04-03T08:30:21.715856Z",
+     "iopub.status.busy": "2023-04-03T08:30:21.714894Z",
+     "iopub.status.idle": "2023-04-03T08:30:21.742610Z",
+     "shell.execute_reply": "2023-04-03T08:30:21.741688Z"
+    },
+    "papermill": {
+     "duration": 0.037798,
+     "end_time": "2023-04-03T08:30:21.744919",
+     "exception": false,
+     "start_time": "2023-04-03T08:30:21.707121",
+     "status": "completed"
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "onnx_models = []\n",
+    "for model in sklearn_models:\n",
+    "    onnx_model = to_onnx(model, X = X_train)\n",
+    "    onnx_models.append(onnx_model)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "005e656f",
+   "metadata": {
+    "execution": {
+     "iopub.execute_input": "2023-04-03T08:30:21.760205Z",
+     "iopub.status.busy": "2023-04-03T08:30:21.759902Z",
+     "iopub.status.idle": "2023-04-03T08:30:21.774192Z",
+     "shell.execute_reply": "2023-04-03T08:30:21.773077Z"
+    },
+    "papermill": {
+     "duration": 0.024592,
+     "end_time": "2023-04-03T08:30:21.776553",
+     "exception": false,
+     "start_time": "2023-04-03T08:30:21.751961",
+     "status": "completed"
+    },
+    "tags": []
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Do sklearn and onnx PCA models generate close results?\n",
+      "True\n",
+      "Do sklearn and onnx KernelPCA models generate close results?\n",
+      "True\n",
+      "Do sklearn and onnx TruncatedSVD models generate close results?\n",
+      "False\n"
+     ]
+    }
+   ],
+   "source": [
+    "for i, model_name in enumerate([\"PCA\", \"KernelPCA\", \"TruncatedSVD\"]):\n",
+    "    sklearn_transformed = sklearn_models[0].transform(X_test)\n",
```

Comment:
Should be `i`, not `0`?

Reply:
@AlibiZhenis I was referring these comments.

Reply:
I think these comments were already addressed [here](https://github.com/opensearch-project/opensearch-py-ml/pull/134/commits/f2b1cb09ef470e9caf7c1940edf6dc2f089f291c)